nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -181,23 +183,52 @@ public IcebergTable table(TableIdentifier tableIdentifier) {
   }
 
   public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    getRef().checkMutable();
+
+    if (namespace.isEmpty()) {
+      throw new IllegalArgumentException("Creating empty namespaces is not supported");
+    }
+
     try {
-      getRef().checkMutable();
-      withReference(
-              getApi()
-                  .createNamespace()
-                  .namespace(org.projectnessie.model.Namespace.of(namespace.levels()))
-                  .properties(metadata))
-          .create();
-      refresh();
-    } catch (NessieNamespaceAlreadyExistsException e) {
-      throw new AlreadyExistsException(e, "Namespace already exists: %s", namespace);
-    } catch (NessieNotFoundException e) {
+      org.projectnessie.model.Namespace content =
+          org.projectnessie.model.Namespace.of(metadata, namespace.levels());
+      ContentKey key = content.toContentKey();
+
+      CommitMultipleOperationsBuilder commitBuilder =
+          api.commitMultipleOperations()
+              .commitMeta(NessieUtil.buildCommitMetadata("create namespace " + key, catalogOptions))
+              .operation(Operation.Put.of(key, content));
+
+      Tasks.foreach(commitBuilder)
+          .retry(5)
+          // conflict can be thrown when parent namespace doesn't exist;
+          // bad request can be thrown when namespace already exists.
+          .stopRetryOn(
+              NessieNotFoundException.class,
+              NessieConflictException.class,
+              NessieBadRequestException.class)
+          .throwFailureWhenFinished()
+          .onFailure((o, exception) -> refresh())
+          .run(
+              b -> {
+                Branch branch = b.branch((Branch) getRef().getReference()).commit();
+                getRef().updateReference(branch);
+              },
+              BaseNessieClientServerException.class);
+
+    } catch (NessieReferenceNotFoundException e) {
       throw new RuntimeException(
           String.format(
-              "Cannot create Namespace '%s': " + "ref '%s' is no longer valid.",
+              "Cannot create Namespace '%s': ref '%s' is no longer valid.",
               namespace, getRef().getName()),
           e);
+    } catch (Exception e) {
+      if (e instanceof NessieBadRequestException
```

Comment:
Note: unfortunately the situation when a create operation fails because it conflicts with an existing key does not surface with a `NessieConflictException` but with a `NessieBadRequestException`. We are forced to parse the message to know if it's a key conflict or something else.

Reply:
For the record, I brought up the issue in Nessie's Zulip chat:

https://project-nessie.zulipchat.com/#narrow/stream/371193-dev/topic/Bad.20Request.20vs.20Conflict.20when.20content.20exists/near/397099981